### ISSUE TITLE
Frange Geodist

### ISSUE THREAD

**Issue by FabianKoestring** [solariumphp/solarium#229]
Hey there!

I try to create a query like mentioned in the solr docs.
http://wiki.apache.org/solr/SpatialSearch#How_to_facet_by_distance

``` php
$client = $this->getServiceManager()->get('Solarium\Client');
$query = $client->createSelect();
$query->setQuery($this->createQuery());
$query->setRows(0);
$query->addParam('pt','85.2,8');
$query->addParam('sfield','distance');
$facetSet = $query->getFacetSet();
$facetSet->setMinCount(1);
$facet = $facetSet->createFacetMultiQuery('distance');
$facet->createQuery('1','{!frange l=0 u=0.1}geodist()');
$facet->createQuery('2','{!frange l=0 u=10}geodist()');
$resultset = $client->select($query);
```

But i always get the exception:

```
geodist - not enough parameters
```

Can you help me?

**Comment by sunthera** [comment]
@Spodnet thanks for the hack!

**Comment by mollux** [comment]
@sunthera @Spodnet can you give a more complete example how you worked around this without patching solarium? or isn't this possible without patch?

When I try the example of @Spodnet , I only get a key in the facet query, and no facets are returned.

**Comment by sunthera** [comment]
@mollux it's because solarium tries to map facet query results to non-existing keys:
`$facetSet->createFacetQuery('distance5 frange l=0 u=5')->setQuery('geodist()');` creates key with name `distance5 frange l=0 u=5`, but solr response contains key name `distance5`. Dump your response result from solr to see it.

I created kludge plugin for this:

```php

class SolariumPlugin extends AbstractPlugin
{
    protected function initPluginType()
    {
        $dispatcher = $this->client->getEventDispatcher();

        $dispatcher->addListener(Events::POST_CREATE_REQUEST, [$this, 'fixGeodistKeyName']);
    }

    /**
     * Change keys from 'distance_0.5 frange l=0 u=0.501' to 'distance_0.5'
     *
     * @param PostCreateRequest $event
     */
    public function fixGeodistKeyName(PostCreateRequest $event)
    {
        /** @var FacetSet $facetSet */
        $facetSet = $event->getQuery()->getFacetSet();
        $facets = $facetSet->getFacets();

        foreach ($facets as $facet) {
            $facetKey = $facet->getKey();

            $pattern = '/(?<key>distance_[0-9]{1,2}(\.[0-5]{1,2})?)/';

            if (preg_match($pattern, $facetKey, $match)) {
                $facetSet->removeFacet($facet);
                $facet->setKey($match['key']);
                $facetSet->addFacet($facet);
            }

        }
    }
```

**Comment by thomascorthals** [comment]
Fixed in [Solarium 6.1.4](https://github.com/solariumphp/solarium/releases/tag/6.1.4).
